reports/aksVersions/aks-versions.sh

```diff
@@ -0,0 +1,147 @@
+# repository running on the cluster.
+#
+# Chart information from the namespaces listed below would be extracted and processed.
+# - admin
+# - monitoring
+# - flux-system
+#
+# Helm whatup would also be using in conjunction with some scripting functionality to
+# archive the desired aim. Data would be transformed and stored in cosmosdb
+# Steps/Flows
+# ----------
+# 1. Get list of charts in helmrepositories, read and extract chart name and url, filter by admin type namespace
+#    and add charts to helm
+# 2. Use helm whatup to get installed and current version the make a verdict. Update the document with additional data
+# 3. Save document generated to cosmosdb with the aid of a python script
+#############################################################################
+
+# Suppressing Intellij specific IDE syntax warning
+# -- PLEASE IGNORE BELOW --
+# shellcheck disable=SC2004
+# shellcheck disable=SC2028
+# shellcheck disable=SC2046
+# shellcheck disable=SC2059
+# shellcheck disable=SC2153
+# --
+
+# ---------------------------------------------------------------------------
+# Define environment variables
+# ---------------------------------------------------------------------------
+clientId=$AZURE_CLIENT_ID
+clientSecret=$AZURE_CLIENT_SECRET
+tenantId=$AZURE_TENANT_ID
+
+# ---------------------------------------------------------------------------
+# Define functions
+# ---------------------------------------------------------------------------
+# ---------------------------------------------------------------------------
+# Make a REST call to cosmos
+# Document is handed over to a python process. Much easier using python that bash
+# for connecting and saving to cosmos.
+# ---------------------------------------------------------------------------
+store_document() {
+  # Add uuid, created date and environment info
+  python3 ./save-to-cosmos.py "${1}"
+  wait $!
+}
+
+echo "Job process start"
+# ---------------------------------------------------------------------------
+# STEP 1:
+# ---------------------------------------------------------------------------
+# Get a list of subscriptions
+# --------------------------------------------------------------------------
+declare -a subscriptions
+
+sharedServicesSubs=$(az account list --query "[?contains(name,'SHAREDSERVICES')].{SubscriptionName:name, SubscriptionID:id, TenantID:tenantId}" --output json | jq -r '.[].SubscriptionID')
+[[ "$sharedServicesSubs" == "" ]] && echo "Error: cannot get a list of Shared Service subscriptions." && exit 1
+
+for sub in $(echo "$sharedServicesSubs"); do
+  subscriptions+=($sub)
+done
+
+cftSubs=$(az account list --query "[?contains(name,'CFT')].{SubscriptionName:name, SubscriptionID:id, TenantID:tenantId}" --output json | jq -r '.[].SubscriptionID')
+[[ "$cftSubs" == "" ]] && echo "Error: cannot get a list of CFT subscriptions." && exit 1
+
+for sub in $(echo "$cftSubs"); do
+  subscriptions+=($sub)
+done
+
```

Comment:
```suggestion
subscriptions=$(az account list --query "[?contains(name, 'CFT') || contains(name, 'SHAREDSERVICES')].{SubscriptionName:name, SubscriptionID:id, TenantID:tenantId}" --output json | jq -r '.[].SubscriptionID')
[[ "$subscriptions" == "" ]] && echo "Error: cannot get a list of subscriptions." && exit 1

```

I think this could be improved a bit to make it easier to read -- as far I can tell you aren't using CFT/SDS further down but just here separately to collate subscriptions, maybe something like this instead? I don't think you need the two for loops then either

Reply:
Something along those code lines anyway^

Reply:
Fair